### PR TITLE
Add json api guide

### PR DESCRIPTION
**Pull request by aklish**

### CODE REVIEW

File: pages/guide/07-jsonapi.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+## Hierarchical URLs
+
+Elide generally follows the [JSON-API recommendations](http://jsonapi.org/recommendations/) for URL design.
+
+There are a few caveats given that Elide allows developers control over how entities are exposed:
+
+1. Some entities may only be reached through a relationship to another entity.  Not every entity is _rootable_.
+1. The root path segment of URLs are by default the name of the class (lowercase).  This can be overridden.
+1. Elide allows relationships to be nested arbitrarily deep in URLs.
+1. Elide currently requires all individual entities to be addressed by ID within a URL.  For example, consider a model with an article with a singular author which has a singular address.   While unambiguous, the following is *not* allowed: `/articles/1/author/address`.  Instead, the author must be full qualified by ID: `/articles/1/author/34/address`
```
Comment (shadanan) [inline pages/guide/07-jsonapi.md]: must be _fully_ qualified

File: pages/guide/07-jsonapi.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+Filters are only supported on attributes with simple, primitive types.
+
+### Rough BNF Syntax
+
+Filters will be passed as query parameters in the URL with the following BNF syntax:
+
+```
+<QUERY> ::= 
+     “filter” "[" <TYPE> "." <ATTRIBUTE> “]“ “=” <VALUES>
```
Comment (shadanan) [inline pages/guide/07-jsonapi.md]: there are mixed quote styles here "" vs “” maybe due to copy / paste

File: pages/guide/07-jsonapi.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+`/articles/1/comments?include=comments.votes&filter[comment.title][in]=Elide%20is%20great&filter[vote.up]=true` 
+
+## Model Identifiers
+
+Elide supports three mechanisms by which a newly created entity is assigned an ID:
+
+1. The ID is assigned by the client and saved in the data store.
+1. The client doesn't provide an ID and the data store generates one.
+1. The client provides an ID which is replaced by one generated by the data store.  When using the patch extension, the client
```
Comment (shadanan) [inline pages/guide/07-jsonapi.md:92]: Are there any restrictions in the format of the ID? According to json-api doc, it's supposed to be a UUID.

**Comment by shadanan** [comment]
:+1:

**Comment by yahoocla** [comment]
CLA is valid!

**Pull request merged**